src/muse.c

```diff
@@ -1957,9 +1967,8 @@ mloot_container(
 
     if (!container || !Has_contents(container) || container->olocked)
         return res; /* 0 */
-    /* FIXME: handle cursed bag of holding */
     if (Is_mbag(container) && container->cursed)
-        return res; /* 0 */
+        boh_loss(mon, container, container->ocarry == mon);
```

Comment:
Is there anything to stop a monster from repeatedly looking into a bag of holding without "learning" it's cursed and losing all the items?

Reply:
No. It's a similar problem to monsters not being able to remember whether something is locked.

Reply:
Since monsters implicitly know items as if they were fully identified, shouldn't they respond to a cursed bag of holding by dumping out all the contents so as not to lose items looking in each time?